src/sharing/ShareToPm.js

```diff
@@ -88,38 +54,13 @@ class ShareToPm extends React.Component<Props, State> {
     this.setState({ choosingRecipients: false });
   };
 
-  setSending = () => {
-    this.setState({ sending: true });
-  };
-
-  handleChooseRecipients = selectedRecipients => {
+  handleChooseRecipients = (selectedRecipients: $ReadOnlyArray<UserId>) => {
     this.setState({ selectedRecipients });
     this.setState({ choosingRecipients: false });
   };
 
-  handleSend = async () => {
-    const _ = this.context;
-    const { auth } = this.props;
-    const { sharedData } = this.props.route.params;
-    const { selectedRecipients, message } = this.state;
-    const data = { selectedRecipients, message, sharedData, type: 'pm' };
-
-    this.setSending();
-    await handleSend(data, auth, _);
-    this.finishShare();
```

Comment:
This can result in an unhandled promise rejection right? should this be wrapped in a try catch? I am using this inside `handleSend` of the `ShareWrapper` introduced in the current revision as well and haven't wrapped it in try catch just yet, wanted someone's opinion on this.

Reply:
Yeah, thanks for catching that.

In the current revision, it looks like there are two `await`s in `handleSend`. You have a try/catch around the one for `api.sendMessage`, so I think that one is taken care of. It'd be good to add another try/catch around the `await` line that does `api.uploadFile`, too.